common/components/Header/components/GasPriceDropdown.tsx

```diff
@@ -5,25 +5,32 @@ import DropdownShell from 'components/ui/DropdownShell';
 import './GasPriceDropdown.scss';
 import { SetGasLimitFieldAction } from 'actions/transaction';
 import { gasPricetoBase } from 'libs/units';
+import { AppState } from 'reducers';
+import { getGasPrice } from 'selectors/transaction';
+import { connect } from 'react-redux';
 
-interface Props {
-  value: string;
+interface OwnProps {
   onChange(payload: SetGasLimitFieldAction['payload']): void;
 }
 
-export default class GasPriceDropdown extends Component<Props> {
+interface StateProps {
+  gasPrice: AppState['transaction']['fields']['gasPrice'];
+}
+
+type Props = OwnProps & StateProps;
+
+class GasPriceDropdown extends Component<Props> {
   constructor(props: Props) {
     super(props);
     this.updateGasPrice = throttle(this.updateGasPrice, 50);
   }
 
   public render() {
-    const { value } = this.props;
     return (
       <DropdownShell
         color="white"
         size="smr"
-        ariaLabel={`adjust gas price. current price is ${value} gwei`}
+        ariaLabel={`adjust gas price. current price is ${this.props.gasPrice.raw} gwei`}
```

Comment:
can we keep the de-structure?

Reply:
I can revert, but this prevents creating another variable that's only going to be used once anyway.

Reply:
It get's used quite a bit in that component afaict.

Reply:
Nevermind, across multiple methods. No real benefit.